base/set.jl

```diff
@@ -1,5 +1,33 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+"""
+	Set{T} <: AbstractSet{T}
+
+The default implementation of hash sets.
+
+`Set`s are containers that provide fast lookup and membership testing.
+Elements in a `Set` are unique, as determined by the element's implementation of
+`isequal` and `hash`. Adding an element that already is in a `Set` does nothing.
```

Comment:
This comment may be misleading. How to define an element that is already in a `Set`? By `isequal` or `in`, respectively? Then you can replace `1` by `1.0`, so the adding (pushing into) does something, but the number of elements remains the same.

``` julia
julia> s = Set()
Set{Any}()

julia> push!(s, 1)
Set{Any} with 1 element:
  1

julia> push!(s, 1.0)
Set{Any} with 1 element:
  1.0

```

Reply:
You are right, this was wrong. I've removed the statement, since the actual behaviour of adding two objects with the same hash is presumably an implementation detail.

Reply:
I can't find this corner case to be documented. So what about this?
"Adding an element that already is `in` the `Set` replaces the old element."

Reply:
An example showing that `push!` is how you add elements might also be good. So far the fact that the Set is mutable isn't described or shown.